working/augmentation-libraries/feature-specification.md

```diff
@@ -616,6 +626,59 @@ It's a **compile-time** error if:
     other top-level variables that might be in scope in the library
     augmentation.*
 
+### Augmenting function and constructor signatures
+
+[signature matching]: #Augmenting_function_and_constructor_signatures
+
+When augmenting a function (top level, static method, instance method, etc.) or
+constructor (generative, factory, etc.) the parameter lists must be the same in
+all meaningful ways. We say that an augmenting function or constructor's
+signature *matches* if:
+
+*   It has the same number of type parameters with the same type parameter names
+    (same identifiers) and bounds (after type annotation inheritance), if any
+    (same *types*, even if they may not be written exactly the same in case one
+    of the declarations needs to refer to a type using an import prefix).
+
+*   The return type (after type annotation inheritance) is the same as the
+    augmented declaration's return type.
+
+*   It has the same number of positional and optional parameters as the
+    augmented declaration.
+
+*   It has the same set of named parameter names as the augmented declaration.
+
+*   For all corresponding pairs of parameters:
+
+    *   They have the same type (after type annotation inheritance).
+
+    *   They have the same `required` and `covariant` modifiers.
+
+    *   If augmenting a constructor, they must both be initializing formals,
+        super parameters, or neither.
```

Comment:
No longer relevant. If a constructor has a super parameter or initializing parameter, it's complete and can't be augmented with implementation. (If it us agumented, it should be by an incomplete/abstract signature-only declaration that only adds metadata. That declaration should not have to repeat initializing parameters. Rather, it *must not*.)

Reply:
Oh, right! Added a paragraph explaining this.